src/library/ButtonGroup/ButtonGroup.js

```diff
@@ -0,0 +1,294 @@
+/* @flow */
+import React, { Children, Component, cloneElement } from 'react';
+import { createStyledComponent } from '../styles';
+import { setFromArray, toArray } from '../utils/collections';
+import composeEventHandlers from '../utils/composeEventHandlers';
+
+type Props = {
+  /** Accessible label */
+  'aria-label': string,
+  /**
+   * Index or array of indices of the selected [Button](/components/button).
+   * Primarily for use with controlled components with a `mode` prop defined.
+   * If this prop is specified, an `onClick` handler must also be specified.
+   * See also: `defaultChecked`
+   */
+  checked?: number | Array<number>,
+  /** Mineral [Button](/components/button) components */
+  children: React$Node,
+  /**
+   * Index or array of indices of the selected [Button(s)](/components/button);
+   * primarily for use with uncontrolled components with a `mode` prop defined.
+   */
+  defaultChecked?: number | Array<number>,
+  /** Disable all [Button](/components/button) children */
+  disabled?: boolean,
+  /** Stretch ButtonGroup to fill its container */
+  fullWidth?: boolean,
+  /** Behavioral mode of [Button](/components/button) children */
+  mode?: 'checkbox' | 'radio',
+  /** Called when a toggleable Button is selected */
+  onChange?: (event: SyntheticEvent<HTMLButtonElement>) => void,
+  /** Called with the click event */
+  onClick?: (event: SyntheticEvent<HTMLButtonElement>) => void,
+  /** Available sizes */
+  size?: 'small' | 'medium' | 'large' | 'jumbo',
+  /** Available variants */
+  variant?: 'danger' | 'success' | 'warning'
+};
+
+type State = {
+  checked: Set<number>
+};
+
+export const componentTheme = (baseTheme: Object) => ({
+  ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
+  ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
+  ButtonGroupButton_borderStartColor: baseTheme.borderColor,
+  ButtonGroupButton_borderStartColor_checked: 'currentcolor',
+  ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,
+
+  ...baseTheme
+});
+
+const styles = ({ fullWidth, theme: baseTheme }) => {
+  let theme = componentTheme(baseTheme);
+  const { direction } = theme;
+  const rtl = direction === 'rtl';
+
+  return {
+    display: 'flex',
+    marginBottom: 0,
```

Comment:
There was a lot going on, so I'm likely confused, but I thought we agreed this wasn't necessary.